pkg/clusterd/inventory/node_test.go

```diff
@@ -61,162 +57,64 @@ func TestLoadDiscoveredNodes(t *testing.T) {
-	verifyProcConfig(t, nodeConfig[machineId], procsConfig)
-	verifyMemoryConfig(t, nodeConfig[machineId], memConfig)
-	verifyNetworkConfig(t, nodeConfig[machineId], netsConfig)
-	assert.Equal(t, "10.0.0.43", nodeConfig[machineId].PublicIP)
-	assert.Equal(t, "root=default,dc=datacenter1", nodeConfig[machineId].Location)
-}
+	cfg := nodeConfig[nodeID]
 
-func setProcInfo(etcdClient *util.MockEtcdClient, procsKey string, procId uint, physicalId uint, siblings uint,
-	coreID uint, numCores uint, speed float64, bits uint) ProcessorConfig {
-
-	procKey := path.Join(procsKey, strconv.FormatUint(uint64(procId), 10))
-	etcdClient.Set(ctx.Background(), procKey, "", &etcd.SetOptions{Dir: true})
-
-	etcdClient.Set(ctx.Background(), path.Join(procKey, ProcPhysicalIDKey), strconv.FormatUint(uint64(physicalId), 10), nil)
-	etcdClient.Set(ctx.Background(), path.Join(procKey, ProcSiblingsKey), strconv.FormatUint(uint64(siblings), 10), nil)
-	etcdClient.Set(ctx.Background(), path.Join(procKey, ProcCoreIDKey), strconv.FormatUint(uint64(coreID), 10), nil)
-	etcdClient.Set(ctx.Background(), path.Join(procKey, ProcNumCoresKey), strconv.FormatUint(uint64(numCores), 10), nil)
-	etcdClient.Set(ctx.Background(), path.Join(procKey, ProcSpeedKey), strconv.FormatFloat(speed, 'f', 2, 64), nil)
-	etcdClient.Set(ctx.Background(), path.Join(procKey, ProcBitsKey), strconv.FormatUint(uint64(bits), 10), nil)
-
-	return ProcessorConfig{
-		ID:         procId,
-		PhysicalID: physicalId,
-		Siblings:   siblings,
-		CoreID:     coreID,
-		NumCores:   numCores,
-		Speed:      speed,
-		Bits:       bits,
-	}
-}
+	// verify the single disk (the partition is not saved)
+	assert.Equal(t, 1, len(cfg.Disks))
+	assert.True(t, cfg.Disks[0].Available)
+	assert.Equal(t, d1.Rotational, cfg.Disks[0].Rotational)
+	assert.Equal(t, d1.Size, cfg.Disks[0].Size)
+	assert.Equal(t, d1.Type, cfg.Disks[0].Type)
 
-func setMemoryInfo(etcdClient *util.MockEtcdClient, memKey string, totalMem uint64) MemoryConfig {
-	etcdClient.Set(ctx.Background(), path.Join(memKey, MemoryTotalSizeKey), strconv.FormatUint(totalMem, 10), nil)
-	return MemoryConfig{TotalSize: totalMem}
-}
-
-func setNetInfo(etcdClient *util.MockEtcdClient, netsKey string, name string, ipv4Addr string, ipv6Addr string, speed uint64) NetworkConfig {
-	netKey := path.Join(netsKey, name)
-	etcdClient.Set(ctx.Background(), netKey, "", &etcd.SetOptions{Dir: true})
-
-	etcdClient.Set(ctx.Background(), path.Join(netKey, NetworkIPv4AddressKey), ipv4Addr, nil)
-	etcdClient.Set(ctx.Background(), path.Join(netKey, NetworkIPv6AddressKey), ipv6Addr, nil)
-	speedStr := ""
-	if speed > 0 {
-		speedStr = strconv.FormatUint(speed, 10)
-	}
-	etcdClient.Set(ctx.Background(), path.Join(netKey, NetworkSpeedKey), speedStr, nil)
-
-	return NetworkConfig{
-		Name:        name,
-		IPv4Address: ipv4Addr,
-		IPv6Address: ipv6Addr,
-		Speed:       speed,
-	}
-}
+	// verify the processors
+	assert.Equal(t, 3, len(cfg.Processors))
```

Comment:
do we need to be verifying the properties of the procs in addition to the count?